### ISSUE TITLE
[QUESTION/HELP] ZERO3 get weight participate in loss

### ISSUE THREAD

**Issue by lxt160980** [deepspeedai/DeepSpeed#7464]
Hi, i am trying to engage part of the weights in my loss calculation, say something like l2norm, and i am currently using zero-stage3
However, after initialization, model_engine.parameters seem to just be placeholders until loss.backward
I tried using l2_norm(safe_get_full_fp32_param(params）) but this function seems to only provide copied buffer, thus no gradient derives from this operation.
safe_set_full_grad would work but that would be involving tons of manual calculation and rule-setting, also breaking the code guideline.
What should i do? Would using GatheredParams context work? I am still struggling here so it would really be a lot of help if you could provide some suggestions.
Thanks!

**Comment by lxt160980** [comment]
I tried using GatheredParams in my loss calculation , when i use safe_get_grad the gradient changed, but the real gradient applied to parameters stayed the same.
Here's my code snippet.
```
# in loss function
with deepspeed.zero.GatheredParameters(params_to_norm) as gathered_params:
# self.model = model_engine
      for name, param in self.model.module.named_parameters():
               # param is of shape() , like tensor(0.7)
                loss += param ** 2
...
# outside loss function
model_engine.backward(loss)
logger.info(deepspeed.utils.safe_get_full_grad(certain_param))
model_engine.step()
```
